### DISCUSSION TITLE
inject database

### DISCUSSION THREAD

**Discussion by bluebrown** [go-swagger/go-swagger#2689] [Q&A]
Hi, I am wondering where to inject a database object. Do I need to create a custom server for that, or is there an easier way?

The DB should be configured somewhere in the top level and connect. Then a pointer to the DB struct should be available to use it in the handlers.

In the KV store example I can see that this is solved through a custom main function in the cmd folder, but I am not sure if that's the only way to do it.

**Comment by casualjim** [comment]
You can use a custom server for that, or pass it in via a factory function for your handlers.

Another alternative is to set up a global middleware and add the database onto the request context early on

**Reply by bluebrown** [reply]
Yeah, the factory way is what they did in the kv store example. But they inject it into the factory from a custom main function. 

The only other option I see is to do it in configure_myapp.go since all the default handler are there. So inside configureAPI one could initialize the database, I guess.
